crates/swc_plugin_runner/src/imported_fn/mod.rs

```diff
@@ -0,0 +1,136 @@
+//! Functions for syntax_pos::hygiene imported into the guests (plugin) runtime
+//! allows interop between host's state to plugin. When guest calls these fn,
+//! it'll be executed in host's memory space.
+/*
+* Below diagram shows one reference example how guest does trampoline between
+* host's memory space.
+* ┌───────────────────────────────────────┐    ┌─────────────────────────────────────────────┐
```

Comment:
small diagram to understand which one lives where, as I myself keep forgetting this 😅